movieland_persistence_jdbc/src/main/java/com/solovey/movieland/dao/jdbc/QueryGenerator.java

```diff
@@ -19,4 +20,14 @@ public String addSorting(String sql, Map<String, SortDirection> sortType) {
         }
         return sql;
     }
+
+    public String buildMoviesReportQuery(String sql, ReportType reportType) {
+        if (reportType == ReportType.ADDED_DURING_PERIOD) {
+            return sql + " where m.added_date between ? and ?" +
+                    " group by m.movie_id";
+        } else {
+            return sql + " group by m.movie_id";
+        }
+
+    }
```

Comment:
1. Would be better replace question marks with named parameters
2. Method contains duplicate code. Where clause can be added in IF and GROUP BY can be moved outside of the condition block because it's added in any case